src/lib/core/BUILD.gn

```diff
@@ -104,6 +104,12 @@ source_set("error") {
   ]
 }
 
+source_set("encoding") {
```

Comment:
Is it necessary to remove the circular dependency between core and support? Wouldn't a proper solution be moving `CHIPEncoding.h` to `support/` instead of having a single directory that contains kinda independent build modules?
This could be too much work though so the current solution is fine with me, but I just want to understand the rationale for the change.